src/analysis/ExpressionExplorer.jl

```diff
@@ -988,9 +992,15 @@ function maybe_macroexpand(ex::Expr; recursive = false, expand_bind = true)
     end
 
     if recursive && (result isa Expr)
-        Expr(result.head, maybe_macroexpand.(result.args; recursive = recursive, expand_bind = expand_bind)...)
```

Comment:
Getting rid of broadcasting is totally fine if it speeds stuff up, go for loops!
BUT, did you see how it performed with `map(result.args) do x maybe_macroexpand(x, recursive=recursive, expand_bind=expand_bind) end`? Just curious if that takes similar compilation-related performance hits

Reply:
> Just curious if that takes similar compilation-related performance hits

Noice. Well spotted. `map` is fine too.

Benchmarks below. All after restarting Julia with `julia --project -ie 'using Pluto'`.

## with broadcasting (original)

```julia
julia> @time @eval Pluto.ExpressionExplorer.maybe_macroexpand(:(@time 1));
  0.306775 seconds (894.16 k allocations: 48.205 MiB, 22.07% gc time, 99.59% compilation time)
```

I think that this is so bad because the types are unknown and multiple types show up in various calls in the recursion which need to be compiled just-in-time. Also, broadcasting takes long to compile. See `@code_typed identity.([1, 2])` for more information.

## with `map`

```julia
julia> @time @eval Pluto.ExpressionExplorer.maybe_macroexpand(:(@time 1));
  0.079647 seconds (86.74 k allocations: 4.639 MiB, 101.12% compilation time)
```

## with a loop

```julia
julia> @time @eval Pluto.ExpressionExplorer.maybe_macroexpand(:(@time 1));
  0.090846 seconds (87.07 k allocations: 4.655 MiB, 99.57% compilation time)
```